### PR TITLE
Remove logging from function and structs.

### LINKED ISSUE #270: Warn about unknown keys found in config.
Got bit by this while debugging the new filter feature, I used "whitelist" instead of the expected "white_list". We should at least warn about unknown keys so users aren't expecting something like this to take when it's an unexpected key.

### PR DESCRIPTION
**Pull request by shawn-hurley**
**Describe what this PR does and why we need it**:
Removes logging from the function signatures and structures
Changes proposed in this pull request
 - create a utility/logging package to manage the logging.
 - adds `log` var to each package that needs it that is package accessible.
 - removes log from functions and structs

Most of the fixes are in util/logging/logger.go. All the other changes are setting up the logger per package and removing the log from structures and functions

**Which issue this PR fixes (This will close that issue when PR gets merged)**
fixes #270

### CODE REVIEW

File: pkg/apb/types.go [hunk 1]
```diff
@@ -19,11 +19,13 @@ package apb
 import (
 	"encoding/json"
 
-	logging "github.com/op/go-logging"
 	"github.com/openshift/ansible-service-broker/pkg/config"
+	utillogging "github.com/openshift/ansible-service-broker/pkg/util/logging"
```
Comment (jmrodri) [inline pkg/apb/types.go]: After reading it a few times, how about 'logutil` instead of `utillogging`? :)

File: pkg/app/app.go [hunk 2]
```diff
@@ -153,84 +153,86 @@ func CreateApp() App {
 		os.Stderr.WriteString(err.Error() + "\n")
 		os.Exit(1)
 	}
-	fmt.Printf("%#v", app.config)
-
-	if app.log, err = NewLog(app.config); err != nil {
+	c := utillogging.LogConfig{
```
Comment (jmrodri) [inline pkg/app/app.go]: logutil instead of utillogging

File: pkg/util/logging/logger.go [hunk 3]
```diff
@@ -33,47 +32,23 @@ type LogConfig struct {
...
-	config := LogConfig{
-		LogFile: c.GetString("log.logfile"),
-		Stdout:  c.GetBool("log.stdout"),
-		Level:   c.GetString("log.level"),
-		Color:   c.GetBool("log.color"),
-	}
+var backends []logging.Backend
+var logLevel logging.Level
 
+//InitilizeLog - initilize the logging utility
```
Comment (jmrodri) [inline pkg/util/logging/logger.go]: space after the //

File: pkg/util/logging/logger.go [hunk 4]
```diff
@@ -33,47 +32,23 @@ type LogConfig struct {
...
+func InitilizeLog(config LogConfig) error {
```
Comment (jmrodri) [inline pkg/util/logging/logger.go]: TYPO: InitilizeLog -> InitializeLog

File: pkg/auth/auth.go [hunk 5]
```diff
@@ -24,10 +24,12 @@ import (
 	"path"
 	"strings"
 
-	logging "github.com/op/go-logging"
 	"github.com/openshift/ansible-service-broker/pkg/config"
+	utillogging "github.com/openshift/ansible-service-broker/pkg/util/logging"
 )
 
+var log = utillogging.NewLog()
```
Comment (jmrodri) [inline pkg/auth/auth.go]: Ah so each package does this call. I was searching for where the hell the logger was set. Found it.

File: pkg/util/logging/logger.go [hunk 6]
```diff
@@ -22,7 +22,6 @@ import (
 	"os"
 
 	logging "github.com/op/go-logging"
-	"github.com/openshift/ansible-service-broker/pkg/config"
 )
```
Comment (jmrodri) [inline pkg/util/logging/logger.go]: The package of this file is `package app` yet it lives in `util/logging` which is the right package?

**Comment by rthallisey** [comment]
👍 pending Zeus' comment

**Pull request merged**